tests/ui/field_reassign_with_default.rs

```diff
@@ -0,0 +1,75 @@
+#![warn(clippy::field_reassign_with_default)]
+
+#[derive(Default)]
+struct A {
+    i: i32,
+    j: i64,
+}
+
+struct B {
+    i: i32,
+    j: i64,
+}
+
+fn main() {
+    // wrong, produces first error in stderr
+    let mut a: A = Default::default();
```

Comment:
Something I missed in my first review (my apologies): this would currently be linted by `default_trait_access`. Since that lint is pedantic, this one should get preference.

I suggest putting both lints in a `default_trait` module, and having `default_trait_access` check if `field_reassign_with_default` would trigger, and avoid linting in that case.

Reply:
How would you check if a lint would trigger?

Reply:
Usually having two lints in the same module and targeting the same kind of node would make this easier, but in this case even though both lints target expressions, `field_reassign_with_default` needs more context so it works at the block level.

We could maybe leverage this: knowing that this lint will find the relevant expression before `default_trait_access`, we could record the linted span and have `default_trait_access` check if the expression it would lint lies in the same span, bailing out in that case.

Reply:
Do you mean that it needs to be implemented via some sort of a manual mechanism? Eg. LintB borrowing a feature from LintA to check if LintA would trigger, before doing its own work.

Reply:
Yes, there is no mecanism to disable a lint temporarily AFAIK.

In this case I mean to keep track of the spans linted by `field_reassign_with_default` in a collection, and have `default_trait_access` check that collection before linting. This should work given that `check_block` will run before the `check_expr` that would trigger `default_trait_access`.

Reply:
@ebroto ok I thought about it for a while, and **I can't figure out a way to talk from one lint to another.** The purpose is to keep track of spans linted by field_reassign_with_default for default_trait_access to read.

The reason for why I can't get the lints to talk is that the two lints are both implemented for two separate unit-structs (= no data), and the implementations are called from lib.rs. Static globals are also out-of-the-question I would assume. Caching into LateContext seems ugly too. Total merge of the two lints does not solve the communication issue since the backing data structure is still a unit-struct.

It's the last thing missing.

Reply:
Sorry for not being more precise! 

`declare_lint_pass!` is not the only way to create a new pass, you can do that "manually" and then use `impl_lint_pass!`. This way you can add state to the pass, which can be used by both lints.

I've added the changes myself to a new `default` module. I will dig into the last round of review ASAP.